pkg/kn/commands/service_replace.go

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"errors"
+	knserving "github.com/knative/client/pkg/serving"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewServiceReplaceCommand(p *KnParams) *cobra.Command {
+	var editFlags ConfigurationEditFlags
+	serviceReplaceCommand := &cobra.Command{
+		Use:   "replace NAME",
+		Short: "replace a service.",
+		Example: `
+  # Replace a service 's1' with image dev.local/ns/image:v2
+  kn service replace s1 --image dev.local/ns/image:v2
+
+  # Replace a service's envrionment variables
+  kn service replace s1 --env KEY1=NEW_VALUE1 --env NEW_KEY2=NEW_VALUE2
+
+  # Reset a service's resources to default ones
+  # (earlier configured resource requests and limits will be replaced with default)
+  # (earlier configured environment variables will be cleared too if any)
+  kn service replace s1
+
+  # Replace a service's resources as provided
+  kn service replace s1 --requests-cpu 400m --limits-memory 256Mi`,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) != 1 {
+				return errors.New("requires the service name.")
+			}
+			namespace, err := GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+			client, err := p.ServingFactory()
+			if err != nil {
+				return err
+			}
+			service, err := client.Services(namespace).Get(args[0], v1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			config, err := knserving.GetConfiguration(service)
+			if err != nil {
+				return err
+			}
+			err = editFlags.Replace(config)
+			if err != nil {
+				return err
+			}
+			_, err = client.Services(namespace).Update(service)
```

Comment:
... unless it's to get the resourceVersion and blat that onto the *new* Service object we use to update.